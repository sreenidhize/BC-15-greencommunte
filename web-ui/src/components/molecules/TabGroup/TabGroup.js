import React, { useState } from "react";
import List from "@material-ui/core/List";
import DashboardOutlinedIcon from "@material-ui/icons/DashboardOutlined";
import WorkOutlineOutlinedIcon from "@material-ui/icons/WorkOutlineOutlined";
import BookmarkBorderOutlinedIcon from "@material-ui/icons/BookmarkBorderOutlined";
import DescriptionOutlinedIcon from "@material-ui/icons/DescriptionOutlined";
import DateRangeOutlinedIcon from "@material-ui/icons/DateRangeOutlined";
import HelpOutlineOutlinedIcon from "@material-ui/icons/HelpOutlineOutlined";
import PhoneInTalkOutlinedIcon from "@material-ui/icons/PhoneInTalkOutlined";
import SettingsOutlinedIcon from "@material-ui/icons/SettingsOutlined";
import TabItem from "../../atoms/TabItem/TabItem";
import { Divider } from "@material-ui/core";
import { withRouter } from "react-router-dom";

const TabGroup = (props) => {
  const getInitialStateValue = () => {
    const { location } = props;
    if (location.pathname === "/savedJobs") {
      return 2;
    }
    return 1;
  };
  const [value, setValue] = useState(getInitialStateValue());
  const list1 = [
    { icon: <DashboardOutlinedIcon />, label: "Dashboard" },
    { icon: <WorkOutlineOutlinedIcon />, label: "Find Jobs" },
    { icon: <BookmarkBorderOutlinedIcon />, label: "Saved Jobs" },
    { icon: <DescriptionOutlinedIcon />, label: "Practice Test" },
    { icon: <DateRangeOutlinedIcon />, label: "News & Events" },
    { icon: <HelpOutlineOutlinedIcon />, label: "Need Help?" },
    { icon: <PhoneInTalkOutlinedIcon />, label: "Contact Us" },
    { icon: <SettingsOutlinedIcon />, label: "Settings" },
  ];
  const nextPath = (path) => {
    props.history.push(path);
  };
  return (
    <div data-testid="tab-group">
      <List>
        {list1.map((item, index) => {
          if (index === 1) {
            return (
              <TabItem
                index={index}
                value={value}
                setValue={setValue}
                icon={item.icon}
                isClickable={true}
                label={item.label}
                onClick={() => nextPath("/findJobs")}
              />
            );
          }
          if (index === 2) {
            return (
              <TabItem
                index={index}
                value={value}
                setValue={setValue}
                icon={item.icon}
                isClickable={true}
                label={item.label}
                onClick={() => nextPath("/savedJobs")}
              />
            );
          }
          if (index === 4) {
            return <Divider />;
          }
          return (
            <TabItem
              index={index}
              value={value}
              icon={item.icon}
              isClickable={false}
              label={item.label}
            />
          );
        })}
      </List>
    </div>
  );
};

export default withRouter(TabGroup);
