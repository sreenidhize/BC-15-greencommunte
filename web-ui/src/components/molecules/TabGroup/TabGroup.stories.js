import React from 'react';
import TabGroup from './TabGroup';

export default {
    title: 'molecules/TabGroup',
    component: TabGroup,
};

const Template = (args) => (<div style={{width:270}}><TabGroup {...args}  /></div>); 
 
export const Primary = Template.bind({});

Primary.args = {
    
};