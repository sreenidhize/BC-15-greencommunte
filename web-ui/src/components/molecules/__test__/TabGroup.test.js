import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import TabGroup from "../TabGroup/TabGroup";
import { BrowserRouter } from "react-router-dom";

describe("TabGroup test", () => {
    it("Matches the snapshot", () => {
      const { container } = render(
        <BrowserRouter><TabGroup /></BrowserRouter>
      );
      expect(container).toMatchSnapshot();
    });
  
    test("Test TabGroup molecule render", () => {
      const { getByTestId } = render(<BrowserRouter><TabGroup /></BrowserRouter>);
      expect(getByTestId("tab-group")).toBeInTheDocument();
    });
  });