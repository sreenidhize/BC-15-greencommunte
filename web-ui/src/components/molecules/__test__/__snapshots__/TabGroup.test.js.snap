// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabGroup test Matches the snapshot 1`] = `
<div>
  <div
    data-testid="tab-group"
  >
    <ul
      class="MuiList-root MuiList-padding"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-mockItem-1 MuiListItem-gutters MuiListItem-button"
        data-testid="tab-item"
        role="button"
        style="background-color: white;"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root makeStyles-inactiveIcon-4"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v2h-4V5h4M9 5v6H5V5h4m10 8v6h-4v-6h4M9 17v2H5v-2h4M21 3h-8v6h8V3zM11 3H3v10h8V3zm10 8h-8v10h8V11zm-10 4H3v6h8v-6z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root makeStyles-inactiveItem-3"
        >
          <p
            class="MuiTypography-root makeStyles-label-6 MuiTypography-body1"
          >
            Dashboard
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-mockItem-1 MuiListItem-gutters MuiListItem-button"
        data-testid="tab-item"
        role="button"
        style="background-color: rgba(90, 197, 104, 0.15);"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root makeStyles-activeIcon-5"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 6V4h-4v2h4zM4 8v11h16V8H4zm16-2c1.11 0 2 .89 2 2v11c0 1.11-.89 2-2 2H4c-1.11 0-2-.89-2-2l.01-11c0-1.11.88-2 1.99-2h4V4c0-1.11.89-2 2-2h4c1.11 0 2 .89 2 2v2h4z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root makeStyles-activeItem-2"
        >
          <p
            class="MuiTypography-root makeStyles-label-6 MuiTypography-body1"
          >
            Find Jobs
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-mockItem-1 MuiListItem-gutters MuiListItem-button"
        data-testid="tab-item"
        role="button"
        style="background-color: white;"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root makeStyles-inactiveIcon-4"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 3H7c-1.1 0-2 .9-2 2v16l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root makeStyles-inactiveItem-3"
        >
          <p
            class="MuiTypography-root makeStyles-label-6 MuiTypography-body1"
          >
            Saved Jobs
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-mockItem-1 MuiListItem-gutters MuiListItem-button"
        data-testid="tab-item"
        role="button"
        style="background-color: white;"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root makeStyles-inactiveIcon-4"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8 16h8v2H8zm0-4h8v2H8zm6-10H6c-1.1 0-2 .9-2 2v16c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root makeStyles-inactiveItem-3"
        >
          <p
            class="MuiTypography-root makeStyles-label-6 MuiTypography-body1"
          >
            Practice Test
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-mockItem-1 MuiListItem-gutters MuiListItem-button"
        data-testid="tab-item"
        role="button"
        style="background-color: white;"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root makeStyles-inactiveIcon-4"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root makeStyles-inactiveItem-3"
        >
          <p
            class="MuiTypography-root makeStyles-label-6 MuiTypography-body1"
          >
            Need Help?
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-mockItem-1 MuiListItem-gutters MuiListItem-button"
        data-testid="tab-item"
        role="button"
        style="background-color: white;"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root makeStyles-inactiveIcon-4"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 12h2c0-2.76-2.24-5-5-5v2c1.66 0 3 1.34 3 3zm4 0h2c0-4.97-4.03-9-9-9v2c3.87 0 7 3.13 7 7zm1 3.5c-1.25 0-2.45-.2-3.57-.57-.1-.03-.21-.05-.31-.05-.26 0-.51.1-.71.29l-2.2 2.2c-2.83-1.44-5.15-3.75-6.59-6.59l2.2-2.21c.28-.26.36-.65.25-1C8.7 6.45 8.5 5.25 8.5 4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1 0 9.39 7.61 17 17 17 .55 0 1-.45 1-1v-3.5c0-.55-.45-1-1-1zM5.03 5h1.5c.07.88.22 1.75.45 2.58l-1.2 1.21c-.4-1.21-.66-2.47-.75-3.79zM19 18.97c-1.32-.09-2.6-.35-3.8-.76l1.2-1.2c.85.24 1.72.39 2.6.45v1.51z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root makeStyles-inactiveItem-3"
        >
          <p
            class="MuiTypography-root makeStyles-label-6 MuiTypography-body1"
          >
            Contact Us
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-mockItem-1 MuiListItem-gutters MuiListItem-button"
        data-testid="tab-item"
        role="button"
        style="background-color: white;"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root makeStyles-inactiveIcon-4"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root makeStyles-inactiveItem-3"
        >
          <p
            class="MuiTypography-root makeStyles-label-6 MuiTypography-body1"
          >
            Settings
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </ul>
  </div>
</div>
`;
